internal/ingest/ingest.go

```diff
@@ -150,6 +154,24 @@ func (ing *Ingester) Sync(ctx context.Context, peerID peer.ID, peerAddr multiadd
 	go func() {
 		defer close(out)
 
+		isHttp := false
+		if peerAddr != nil {
+			for _, p := range peerAddr.Protocols() {
+				if p.Code == multiaddr.P_HTTP || p.Code == multiaddr.P_HTTPS {
+					isHttp = true
+					break
+				}
+			}
+		}
+		if isHttp {
+			ing.httpProvidersMu.Lock()
+			if ing.httpProviders == nil {
+				ing.httpProviders = make(map[peer.ID]string)
+			}
+			ing.httpProviders[peerID] = peerAddr.String()
```

Comment:
can there be multiple locations for an http provider?

Reply:
Not in this flow since we are only given a single address to sync to. In general yes though.